federation/pkg/federation-controller/configmap/configmap_controller_test.go

```diff
@@ -143,11 +143,18 @@ func TestConfigMapController(t *testing.T) {
 	}
 
 	configmapWatch.Modify(configmap1)
-	updatedConfigMap = GetConfigMapFromChan(cluster1UpdateChan)
-	assert.NotNil(t, updatedConfigMap)
-	assert.Equal(t, configmap1.Name, updatedConfigMap.Name)
-	assert.Equal(t, configmap1.Namespace, updatedConfigMap.Namespace)
-	assert.True(t, util.ConfigMapEquivalent(configmap1, updatedConfigMap))
+	for {
```

Comment:
This fix irks me, because it seems like the occasional double delivery should not be happening.

That said, it seems not worth the effort to fix this for real in light of upcoming work to refactor the federation controllers to use a common framework.